src/OrleansEventSourcing/EventStorage/LogViewAdaptor.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.EventSourcing.Common;
+using Orleans.LogConsistency;
+using Orleans.Runtime;
+using Orleans.Storage;
+
+namespace Orleans.EventSourcing.EventStorage
+{
+
+    /// <summary>
+    /// A log view adaptor that is based on an event store
+    /// </summary>
+    /// <typeparam name="TState">The type of the log view, or grain state.</typeparam>
+    /// <typeparam name="TEvent">The type of the events (usually just object)</typeparam>
+    internal class EventStoreLogViewAdaptor<TState, TEvent> : PrimaryBasedLogViewAdaptor<TState, TEvent, SubmissionEntryWithGuid<TEvent>>
+        where TState : class, new() where TEvent : class
+    {
+        // the provider that created this adaptor
+        private LogConsistencyProvider provider;
+
+        // the confirmed state (= what was read from storage most recently)
+        private TState confirmedState;
+        private int confirmedVersion;
+
+        // the handle to the event stream returned by the event store provider
+        private IEventStreamHandle eventStream;
+
+        // must be constructed lazily because grain reference is not available at construction time
+        private IEventStreamHandle EventStream { get {
+                return (eventStream ??
+                   (eventStream = provider.EventStore.GetEventStreamHandle(
+                       provider.EventStore.DefaultStreamName(Host.GetType(), Services.GrainReference))));
+            }
+        }
+
+        public EventStoreLogViewAdaptor(
+            ILogViewAdaptorHost<TState, TEvent> host, 
+            LogConsistencyProvider provider,
+            TState initialState, 
+            ILogConsistencyProtocolServices services)  : base(host, initialState, services)
+        {
+            this.provider = provider;
+
+            if (this.provider == null)
+            {
+                throw new ArgumentException("provider must be an event store", nameof(provider));
+            }
+        }
+
+        // streamname must be constructed lazily because grain reference is not available at construction time
+        private string streamName
+        {
+            get
```

Comment:
I don't think this is being used at all, so please remove. Otherwise use Pascal casing for properties